modules/administration-guide/pages/installing-che-in-a-restricted-environment.adoc

```diff
@@ -1,26 +1,77 @@
-:navtitle: Installing {prod-short} in restricted environment
+:navtitle: Installing {prod-short} in a restricted environment
 :keywords: administration guide, installing-che-in-a-restricted-environment, installing-che-in-restricted-environment
-:page-aliases: installation-guide:installing-che-in-restricted-environment, installation-guide:installing-che-in-a-restricted-environment
+:page-aliases: installation-guide:installing-che-in-restricted-environment, installation-guide:installing-che-in-a-restricted-environment, preparing-a-restricted-environment, configuring-che-to-run-in-a-restricted-environment
 
-:parent-context-of-installing-che-in-a-restricted-environment: {context}
+:my_internal_registry: __<my_internal_registry>__
+:my_organization: __<my_organization>__
+:my_tag: {my_internal_registry}/{my_organization}/redhat-operator-index:v{ocp4-ver}
 
-[id="installing-{prod-id-short}-in-a-restricted-environment_{context}"]
-= Installing {prod-short} in a restricted environment
-
-:context: installing-{prod-id-short}-in-a-restricted-environment
+[id="installing-{prod-id-short}-in-a-restricted-environment-on-openshift_{context}"]
+= Installing {prod-short} in a restricted environment on OpenShift
 
 By default, {prod-short} workspaces reference various external resources, including container images available in public registries or sample projects hosted on public resources. This section describes how to deploy {prod-short} in a restricted environment where these external resources are not available. This method is also called _offline mode_ or _air-gap scenario_.
 
 .Prerequisites
 
-* You are logged in to the cluster using the developer account.
+* The destination OpenShift Cluster is ready to operate on a restricted nework. See link:https://docs.openshift.com/container-platform/latest/operators/admin/olm-restricted-networks.html[OpenShift documentation - Using Operator Lifecycle Manager on restricted networks].
+
+* The registry pass:a,q[{my_internal_registry}] on the target OpenShift cluster contains a pass:a,q[{my_organization}] namespace.
+
+* The `opm` tool is installed on your workstation. See link:https://docs.openshift.com/container-platform/4.10/cli_reference/opm/cli-opm-install.html[OpenShift documentation - Installing the `opm` CLI].
+
+* The `curl`, `podman` and `yq` tools are installed on your workstation.
+
+.Procedure
+
+. Run the following commands on your workstation from a location with unrestricted network access to mirror the images required for {prod-short} to local podman registry:
++
+[subs="+attributes,+quotes"]
+----
+$ export TAG='{my_internal_registry}/{my_organization}/redhat-operator-index:v{ocp4-ver}'
+$ export PACKAGES=$( \
+    curl -fsSL {prod-olm-catalog-csv-yaml} \
+    | yq -r '.spec.relatedImages[] | .image' \
+    | tr [:space:] , \
+    | head -c -1)
+$ podman login registry.redhat.io
+$ opm index prune \
+    --from-index {prod-olm-catalog-image} \
+    --packages "$PACKAGES" \
+    --tag "$TAG"
```

Comment:
I have issues with the `opm` command

```
opm index prune     --from-index registry.redhat.io/redhat/community-operator-index:v4.10     --packages "$PACKAGES"     --tag "$TAG"WARN[0000] DEPRECATION NOTICE:
Sqlite-based catalogs and their related subcommands are deprecated. Support for
them will be removed in a future release. Please migrate your catalog workflows
to the new file-based catalog format. 
INFO[0000] pruning the index                             packages="[quay.io/eclipse/che-operator@sha256:a5652335698983a310abc8fd0cb6396117975dad4de43afa05d6312b52b20231 quay.io/brancz/kube-rbac-proxy@sha256:b62289c3f3f883ee76dd4e8879042dd19abff743340e451cb59f9654fc472e4f quay.io/che-incubator/configbump@sha256:175ff2ba1bd74429de192c0a9facf39da5699c6da9f151bd461b3dc8624dd532 
[...]
lipse/che-quarkus@sha256:5a08ef7ecd394c78b1002be3ece6afb1fd084f28b0d48319443e01e805057928 quay.io/eclipse/che-rust-1.57@sha256:f22b578677b61ddee2696359b1e26485f62e1484e08550d6e935b50417bb8986 registry.access.redhat.com/ubi8-minimal@sha256:2f6b88c037c0503da7704bccd3fc73cb76324101af39ad28f16460e7bce98324 registry.access.redhat.com/ubi8/ubi-minimal@sha256:eb98682661cc5fe83bbe583af9d55a90d698cfc0d14af5b83e4568c3e26b4516]"
Error: error pulling image: Trying to pull registry.redhat.io/redhat/community-operator-index:v4.10...
Error: Source image rejected: A signature was required, but no signature exists
. exit status 125
Usage:
  opm index prune [flags]

Flags:
  -i, --binary-image opm        container image for on-image opm command
  -c, --container-tool string   tool to interact with container images (save, build, etc.). One of: [docker, podman] (default "podman")
  -f, --from-index string       index to prune
      --generate                if enabled, just creates the dockerfile and saves it to local disk
  -h, --help                    help for prune
  -d, --out-dockerfile string   if generating the dockerfile, this flag is used to (optionally) specify a dockerfile name
  -p, --packages strings        comma separated list of packages to keep
      --permissive              allow registry load errors
  -t, --tag string              custom tag for container image being built

Global Flags:
      --skip-tls   skip TLS certificate verification for container image registries while pulling bundles or index
```

Reply:
Don't check signatures for the community-operator-index. In `/etc/containers/policy.json` set:

```
    "transports": {
        "docker": {
            "registry.access.redhat.com/redhat/community-operator-index": [
                {
                    "type": "insecureAcceptAnything"
                }
            ],
```